mautic/mautic
# Feature clone segments

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | y |
| Automated tests included? | /
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5604 (closed already)
| BC breaks? | N
| Deprecations? | N

[//]: # ( Required: )
#### Description:
New feature of cloning segment has been created. Because edit action was almost the same as clone action, I made some more changes to prevent code duplication.

#### Steps to test this PR:
1. Try to edit segment - make sure it still works properly
2. Try to clone segment
3. Make sure that filters has been cloned as well
4. Make sure that contacts HASN'T been cloned
5. Make sure that cloned segment is not published
6. The alias will get cleared on clone and automatically created as unique alias on save.
7. Clone/Edit -> Apply will get you to the edit form
8. Clone/Edit -> Save will take you to the detail page
9. Test the Clone button on the segment list as well as on detail page
